fix scrolling of urth-viz-table

peller opened the pull request:
refs #121

peller committed:
set overflow, width and height to allow proper scrolling of handsontable refs #121
iron resize behavior not needed.
overflow:hidden workaround for getTrimmingContainer also no longer needed
upgrade to 0.20.2-ibm fork of handsontable

(c) Copyright IBM Corp. 2015

peller commented:
Seems to work for all except one case: when switching from notebook to dashboard view, the table scrolling then shows the problem @lbustelo describes in #121 where the header and rows scroll separately.  It seems that handsontable has to be triggered to redraw, and while it has document onresize handlers to do this, the dashboard switch is undetected.  Resizing the browser window corrects the problem, and I couldn't make it happen by bringing up a notebook or dashboard directly.

parente commented:
From the dashboard side, we could certainly publish events like jupyter does for our three modes: notebook, dashboard layout, dashboard preview. But then your widget would be aware of (if not tightly coupled to) jupyter_dashboards which is a bit odd.

peller commented:
Ok, so classic parent/child layout pattern. It would make sense to have the layout container resize the child, perhaps through some simple interface/method on the table, or an event which could be DOM-based.

peller commented:
NVM. Looks like @dalogsdon used `iron-resize` for this.

https://github.com/jupyter-incubator/dashboards/blob/458d436f22300e488eb329772312616ecfad9682/urth_dash_js/notebook/dashboard-view/polymer-support.js#L9

peller committed:
add iron-resize handler and directly trigger walkontable event handlers to resize the table
when the individual element is resized by the dashboard. Refs #121

(c) Copyright IBM Corp. 2015

peller committed:
inline urth-viz-table.css stylesheet to avoid multiple link tags, implicit shadow dom id reference

(c) Copyright IBM Corp. 2015

lbustelo commented:
#135 contains information about what was tested and what has failed in previous attempts

lbustelo commented:
On chrome:
I ran through the testing that I was doing for #135. It all works good.

On firefox:
The scrolling was behaving correctly, but we still have problems with styling and overflow in firefox. I consider these a separate issue since now it is not about scrolling and might also be a problem with how dashboard determines the height of a cell. See below:

In Dashboard Layout view you can see that there is certainly some weirdness in the height of the table (the black border) but also the rest of the content from the cell is spilling over. The content overflows over the boundaries of the layout cell, and there is no way to get to the resize corner.

![screen shot 2015-12-15 at 8 54 06 am](https://cloud.githubusercontent.com/assets/2474841/11814011/3064f45a-a30a-11e5-8a5d-f54df5d336e1.png)

![screen shot 2015-12-15 at 8 55 18 am](https://cloud.githubusercontent.com/assets/2474841/11813966/e7eeb094-a309-11e5-80da-3f830ee603d6.png)

In general, it seems that the styles for the table cell content is missing 
```
{
   white-space: nowrap
}
```
With that change... it goes from 
![screen shot 2015-12-15 at 9 03 06 am](https://cloud.githubusercontent.com/assets/2474841/11814128/d11433a2-a30a-11e5-876a-e70e45d37505.png)

to
![screen shot 2015-12-15 at 9 02 57 am](https://cloud.githubusercontent.com/assets/2474841/11814132/d7ea779a-a30a-11e5-8c44-261cbac1e4a6.png)

I'll open issues for the above and move ahead with merging.

@parente I want to cherry-pick these changes to the 0.2.x branch but I think they depend on some code that went into master for dashboards. We should coordinate.
